src/snowflake/connector/result_set.py

```diff
@@ -61,45 +61,71 @@ def result_set_iterator(
     Just like ``ResultBatch`` iterator, this might yield an ``Exception`` to allow users
     to continue iterating through the rest of the ``ResultBatch``.
     """
-
-    with ThreadPoolExecutor(prefetch_thread_num) as pool:
-        # Fill up window
-
-        logger.debug("beginning to schedule result batch downloads")
-
-        for _ in range(min(prefetch_thread_num, len(unfetched_batches))):
-            logger.debug(
-                f"queuing download of result batch id: {unfetched_batches[0].id}"
-            )
-            unconsumed_batches.append(
-                pool.submit(unfetched_batches.popleft().create_iter, **kw)
-            )
-
-        yield from first_batch_iter
-
-        i = 1
-        while unconsumed_batches:
-            logger.debug(f"user requesting to consume result batch {i}")
-
-            # Submit the next un-fetched batch to the pool
-            if unfetched_batches:
+    is_fetch_all = kw.pop("is_fetch_all")
+    if is_fetch_all:
+        with ThreadPoolExecutor(prefetch_thread_num) as pool:
+            logger.debug("beginning to schedule result batch downloads")
+            for _ in range(min(prefetch_thread_num, len(unfetched_batches))):
+                logger.debug(
+                    f"queuing download of result batch id: {unfetched_batches[0].id}"
+                )
+                unconsumed_batches.append(
+                    pool.submit(unfetched_batches.popleft().create_iter, **kw)
+                )
```

Comment:
do you still need this code if it's fetching all?

Reply:
I'll try if we still need this, not sure if it will break tests